installer/lib/phx_new/umbrella.ex

```diff
@@ -1,138 +1,54 @@
   ]
 
-  template :ecto, [
-    {:eex,  "phx_umbrella/apps/app_name/lib/repo.ex", "lib/repo.ex"},
-    {:eex,  "phx_ecto/data_case.ex",                  "test/support/data_case.ex"},
-    {:eex,  "phx_ecto/seeds.exs",                     "priv/repo/seeds.exs"},
-    {:keep, "phx_umbrella/apps/app_name/priv/repo/migrations", "priv/repo/migrations"},
-  ]
-
-  template :brunch, [
-    {:text, "assets/brunch/gitignore",       ".gitignore"},
-    {:eex,  "assets/brunch/brunch-config.js", "assets/brunch-config.js"},
-    {:eex,  "assets/brunch/package.json",     "assets/package.json"},
-    {:text, "assets/app.css",                 "assets/css/app.css"},
-    {:text, "assets/phoenix.css",             "assets/css/phoenix.css"},
-    {:eex,  "assets/brunch/app.js",           "assets/js/app.js"},
-    {:eex,  "assets/brunch/socket.js",        "assets/js/socket.js"},
-    {:keep, "assets/vendor",                  "assets/vendor"},
-    {:text, "assets/robots.txt",              "assets/static/robots.txt"},
-  ]
-
-  template :html, [
-    {:eex,  "phx_umbrella/apps/app_name_web/test/controllers/page_controller_test.exs", "test/controllers/page_controller_test.exs"},
-    {:eex,  "phx_umbrella/apps/app_name_web/test/views/layout_view_test.exs",           "test/views/layout_view_test.exs"},
-    {:eex,  "phx_umbrella/apps/app_name_web/test/views/page_view_test.exs",             "test/views/page_view_test.exs"},
-    {:eex,  "phx_umbrella/apps/app_name_web/lib/controllers/page_controller.ex",        "lib/controllers/page_controller.ex"},
-    {:eex,  "phx_umbrella/apps/app_name_web/lib/templates/layout/app.html.eex",         "lib/templates/layout/app.html.eex"},
-    {:eex,  "phx_umbrella/apps/app_name_web/lib/templates/page/index.html.eex",         "lib/templates/page/index.html.eex"},
-    {:eex,  "phx_umbrella/apps/app_name_web/lib/views/layout_view.ex",                  "lib/views/layout_view.ex"},
-    {:eex,  "phx_umbrella/apps/app_name_web/lib/views/page_view.ex",                    "lib/views/page_view.ex"},
-  ]
-
-  template :bare, [
-    {:text,   "static/bare/gitignore", ".gitignore"},
-  ]
-
-  template :static, [
-    {:text,   "assets/bare/gitignore", ".gitignore"},
-    {:text,   "assets/app.css",         "priv/static/css/app.css"},
-    {:append, "assets/phoenix.css",     "priv/static/css/app.css"},
-    {:text,   "assets/bare/app.js",     "priv/static/js/app.js"},
-    {:text,   "assets/robots.txt",      "priv/static/robots.txt"},
-  ]
-
-
-  def app(base_path, opts) do
-    app = opts[:app] || Path.basename(Path.expand(base_path))
-    app_path = Path.join(base_path <> "_umbrella", "apps/#{app}")
-
-    {app, Module.concat([opts[:module] || Macro.camelize(app)]), app_path}
+  def prepare_project(%Project{app: app} = project) when not is_nil(app) do
+    project
+    |> put_app()
+    |> put_web()
+    |> put_root_app()
   end
+  defp put_app(project) do
+    project_path = Path.expand(project.base_path <> "_umbrella")
```

Comment:
just to clarify, I think automatically adding a suffix works well, however it behaves counterintuitively when a given path already exists in the system (e.g. giving `~/src/foo` or `.`, I expect to generate project in these directories)